### DISCUSSION TITLE
Dropping by, checking in!

### DISCUSSION THREAD

**Discussion by dnlmc** [ted-marozzi/mermaid-flow-feedback#13] [General]
Just dropping by as I'm using the editor again for some new mapping work, great to have as a quickstart for sketching out initial graphs! 

I see some active issues, are you still actively developing this & do you have a sense of roadmap of what features you'll include next?

Thanks again for all the work on this!

**Comment by ted-marozzi** [comment]
Hey there!

Great to see you are still using the editor, yes I am still actively working on this in my spare time.
As for the roadmap, I think thats a good idea, I will create a public roadmap in this repo some time this week. In the short term, I am working on an AI feature to kick start diagrams even faster, then the ability to save diagrams within Mermaid Flow. Then I will continue to flesh out and de-bug the flowchart functionality as there are still some issues and missing features. I then plan to start working on another diagram type.

This is of course all subject to change, so let me know if you think there are crucial missing pieces that should be prioritised!
Cheers.

**Reply by dnlmc** [reply]
Nice!  i guess everything needs an AI interface first & foremost these days lol, but excited to check that out!

nothing crucial missing imo at the moment, but interested in the next diagram types you plan to implement!

also I am going to raise an issue on some weird text label bugs i just encountered in the code panel, seemingly adding weird unicode strings when using enter to create newlines etc.  Not critical, usualy just a `Change All Occurrences` to strip them back out, but a bit annoying. 

Lastly, do you know where the core team is with implementing the new renderer (i think you mentioned this before). I think it was supposed to make it easier to reposition nodes or set custom layouts or something? (probably among other things)

**Reply by ted-marozzi** [reply]
Indeed it does aha I got a fair bit of feedback indicating it would be a useful feature and I must say its cool at the very least.

I was thinking sequence would be next on the list but keen to hear from you and the community on what you would prefer!

Thanks for raising that bug, I will get onto it asap, I have been busy with work and other things lately!

Yeah I think you are talking about the elk renderer, I will try and find the issue. Though one potential hacky solution to allow more custom positioning of nodes is to use invisible edges as demonstrated here

https://github.com/ted-marozzi/mermaid-flow-feedback/assets/38032037/9906cd31-4320-49c4-af10-b32a6bd2b836

You can also position nodes with svg css properties such as x y shown here, but the issue is the text doesn't follow, so maybe you could raise an issue for that in the mermaid repo.

<img width="661" alt="Screenshot 2023-06-29 at 12 21 04 pm" src="https://github.com/ted-marozzi/mermaid-flow-feedback/assets/38032037/23fbd165-3833-4666-841c-d29b691d81db">

**Reply by ted-marozzi** [reply]
I couldn't find a tracking issue but elk is a more powerful layout engine which maybe provide more customisable features, here is the limited docs on it: https://mermaid.js.org/syntax/flowchart.html#renderer

**Reply by ted-marozzi** [reply]
Roadmap released [here](https://github.com/users/ted-marozzi/projects/4/views/1)

**Reply by dnlmc** [reply]
@ted-marozzi nice!  & thanks for the info on the elk renderer & wild workarounds lol, interested to see where all this lands!

& as far as preferences for new diagram types, I mostly just adapt flow charts for everything I wanna do with Mermaid lol, but probably the next most likely that I'd use would be:

- ERD
- Gantt
- Quadrant (2x2)
- Mindmaps (if they become more flexible)

wouldn't necessarily prioritize those for my sake tho, cuz I have no actual concrete use cases in mind at the moment

**Reply by ted-marozzi** [reply]
Lovely, thank you for this info! I am keen to start getting onto more diagram types soon :)

**Reply by ted-marozzi** [reply]
I just launched the [Auto Flow](https://www.mermaidflow.app/flowchart), the AI powered feature! 🎉
The feature requires an account so if you sign up let me know and I am happy to award you a bunch of free credits for being the first and most generous donator, without you I may never have bothered to develop Mermaid Flow!

This feature will be a paid feature and as in soon you will be able to buy credits. I intend to use extra features like this to sustain the development of Mermaid Flow. Importantly though the ability to create and export diagrams will remain completely free and accessible without an account.

Currently I receive around one donation a month, which is not really enough to sustain development long term but I am hoping this is a sustainable path forward.

Let me know what you think! :)

https://github.com/ted-marozzi/mermaid-flow-feedback/assets/38032037/38a7358e-69f6-4992-8ba2-c5825f57d228

**Reply by ted-marozzi** [reply]
@dnlmc
